src/cogent3/core/alignment.py

```diff
@@ -1771,6 +1771,11 @@ def dotplot(
         seq1 = self.named_seqs[name1]
         seq2 = self.named_seqs[name2]
 
+        if isinstance(seq1, Aligned):
```

Comment:
add a comment above this as to why doing the `deepcopy`.

`Deep copying Aligned instance to ensure only region specified by Aligned.map is displayed.`